### ISSUE TITLE
Is this shared memory offset in RegularTileIterator not correct?

### ISSUE THREAD

**Issue by xiaocenxiaocen** [NVIDIA/cutlass#125]
https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/transform/threadblock/regular_tile_iterator_pitch_linear.h#L142  
In `cutlass/transform/pitch_linear_thread_map.h`, there is a comment which says `ThreadMap::Delta` is in unit of element.  
But the `access_ptr` here is a pointer of `AccessType`, I think the offset will be incorrect when `ThreadMap::kElementPerAccess` is greater than 1.

**Comment by hwu36** [comment]
Hmm... Which offset do you think is wrong?.  

Here is `AccessType`

https://github.com/NVIDIA/cutlass/blob/4dac7490e6a846893595de352c57f2bf118796c1/include/cutlass/transform/threadblock/regular_tile_iterator_pitch_linear.h#L87

It contains `ThreadMap::kElementsPerAccess` elements.

Both `frag_ptr` and `access_ptr` are casted to this type
https://github.com/NVIDIA/cutlass/blob/4dac7490e6a846893595de352c57f2bf118796c1/include/cutlass/transform/threadblock/regular_tile_iterator_pitch_linear.h#L130

https://github.com/NVIDIA/cutlass/blob/4dac7490e6a846893595de352c57f2bf118796c1/include/cutlass/transform/threadblock/regular_tile_iterator_pitch_linear.h#L136

So, `frag_ptr[1]` or `access_ptr[1]` is `ThreadMap::kElementsPerAccess` elements away from `frag_ptr[0]` or `access_ptr[0]`

**Comment by xiaocenxiaocen** [comment]
https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/transform/threadblock/regular_tile_iterator_pitch_linear.h#L142
line142 should be
```
frag_ptr[idx] = access_ptr[c * ThreadMap::Delta::kContiguous / ThreadMap::kElementsPerAccess];
```
...

**Comment by hwu36** [comment]
Yeah, you are right.  Line 183 has the same problem.  This iterator is only used by 2-stage SIMT kernels which only use `kElementsPerAccess = 1`

Thank you very much!
